File: prepperapp/ios/PrepperApp/Views/SearchResultCard.swift
import SwiftUI

struct SearchResultCard: View {
    let result: SearchResult
    let onTap: () -> Void
    
    // Map module to priority for color coding
    private var priorityColor: Color {
        // Assuming priority is encoded in the result somehow
        // For now, using module name as a proxy
        switch result.module {
        case "critical", "emergency":
            return .red
        case "important", "core":
            return .yellow
        default:
            return .gray
        }
    }
    
    // Determine if this is high priority based on score
    private var isHighPriority: Bool {
        result.score > 0.8
    }
    
    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 16) {
                // Priority Indicator
                Rectangle()
                    .fill(priorityColor)
                    .frame(width: 8)
                
                VStack(alignment: .leading, spacing: 4) {
                    // Title with large, bold font
                    Text(result.title)
                        .font(.title2)
                        .fontWeight(.bold)
                        .foregroundColor(.white)
                        .lineLimit(2)
                        .multilineTextAlignment(.leading)
                    
                    // Summary with good contrast
                    Text(result.summary)
                        .font(.body)
                        .foregroundColor(.gray)
                        .lineLimit(3)
                        .multilineTextAlignment(.leading)
                    
                    // Module/Category indicator
                    HStack {
                        Text(result.module.uppercased())
                            .font(.caption)
                            .fontWeight(.semibold)
                            .foregroundColor(priorityColor)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 2)
                            .background(priorityColor.opacity(0.2))
                            .cornerRadius(4)
                        
                        Spacer()
                        
                        // Relevance score indicator (for debugging)
                        #if DEBUG
                        Text(String(format: "%.2f", result.score))
                            .font(.caption2)
                            .foregroundColor(.gray)
                        #endif
                    }
                }
                
                // Chevron for navigation hint
                Image(systemName: "chevron.right")
                    .foregroundColor(.gray)
                    .imageScale(.small)
            }
            .padding(.leading, 0)
            .padding(.vertical, 16)
            .padding(.trailing, 16)
            .frame(minHeight: 100) // Large tap target
            .background(Color.black.opacity(0.2))
            .cornerRadius(12)
            .contentShape(Rectangle()) // Make entire area tappable
        }
        .buttonStyle(EmergencyButtonStyle())
    }
}

// Custom button style for emergency use
struct EmergencyButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .scaleEffect(configuration.isPressed ? 0.98 : 1.0)
            .opacity(configuration.isPressed ? 0.8 : 1.0)
            .animation(.easeOut(duration: 0.1), value: configuration.isPressed)
    }
}

// MARK: - Supporting Views

struct SearchResultsList: View {
    let results: [SearchResult]
    let onSelectResult: (SearchResult) -> Void
    
    var body: some View {
        ScrollView {
            LazyVStack(spacing: 12) {
                ForEach(results) { result in
                    SearchResultCard(result: result) {
                        onSelectResult(result)
                    }
                }
            }
            .padding(.horizontal)
            .padding(.top, 8)
        }
    }
}

// MARK: - Preview

struct SearchResultCard_Previews: PreviewProvider {
    static let sampleResults = [
        SearchResult(
            doc_id: "1",
            title: "Severe Bleeding Control",
            summary: "Apply direct pressure to the wound immediately. Use a tourniquet if bleeding cannot be controlled with direct pressure. Call emergency services...",
            score: 0.95,
            module: "critical"
        ),
        SearchResult(
            doc_id: "2",
            title: "Hypothermia Treatment",
            summary: "Move to warm shelter. Remove wet clothing. Insulate the entire body. Give warm beverages if conscious. Seek medical attention...",
            score: 0.87,
            module: "important"
        ),
        SearchResult(
            doc_id: "3",
            title: "Water Purification Methods",
            summary: "Boil water for at least 1 minute. Use water purification tablets. Filter through clean cloth and sand. UV sterilization in clear bottles...",
            score: 0.72,
            module: "core"
        )
    ]
    
    static var previews: some View {
        ZStack {
            Color.black
                .ignoresSafeArea()
            
            SearchResultsList(results: sampleResults) { result in
                print("Selected: \(result.title)")
            }
        }
    }
}

File: prepperapp/android/app/src/main/java/com/prepperapp/ui/components/SearchResultCard.kt
package com.prepperapp.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.ripple.rememberRipple
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalHapticFeedback
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.prepperapp.BuildConfig
import com.prepperapp.SearchResult

@Composable
fun SearchResultCard(
    result: SearchResult,
    onTap: () -> Unit,
    modifier: Modifier = Modifier
) {
    val haptic = LocalHapticFeedback.current
    
    // Map module to priority color
    val priorityColor = when (result.module) {
        "critical", "emergency" -> Color.Red
        "important", "core" -> Color.Yellow
        else -> Color.Gray
    }
    
    Card(
        modifier = modifier
            .fillMaxWidth()
            .heightIn(min = 100.dp) // Large tap target
            .clickable(
                interactionSource = remember { MutableInteractionSource() },
                indication = rememberRipple(bounded = true, color = Color.White.copy(alpha = 0.1f))
            ) {
                haptic.performHapticFeedback(androidx.compose.ui.hapticfeedback.HapticFeedbackType.TextHandleMove)
                onTap()
            },
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = Color.Black.copy(alpha = 0.2f)
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(start = 0.dp, top = 16.dp, end = 16.dp, bottom = 16.dp)
        ) {
            // Priority Indicator
            Box(
                modifier = Modifier
                    .width(8.dp)
                    .fillMaxHeight()
                    .background(priorityColor)
            )
            
            Spacer(modifier = Modifier.width(16.dp))
            
            // Content
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                // Title with large, bold font
                Text(
                    text = result.title,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis
                )
                
                // Summary with good contrast
                Text(
                    text = result.summary,
                    fontSize = 16.sp,
                    color = Color.Gray,
                    maxLines = 3,
                    overflow = TextOverflow.Ellipsis
                )
                
                // Module/Category indicator
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Surface(
                        shape = RoundedCornerShape(4.dp),
                        color = priorityColor.copy(alpha = 0.2f)
                    ) {
                        Text(
                            text = result.module.uppercase(),
                            modifier = Modifier.padding(horizontal = 8.dp, vertical = 2.dp),
                            fontSize = 12.sp,
                            fontWeight = FontWeight.SemiBold,
                            color = priorityColor
                        )
                    }
                    
                    // Relevance score (debug only)
                    if (BuildConfig.DEBUG) {
                        Text(
                            text = String.format("%.2f", result.score),
                            fontSize = 11.sp,
                            color = Color.Gray.copy(alpha = 0.6f)
                        )
                    }
                }
            }
            
            Spacer(modifier = Modifier.width(8.dp))
            
            // Navigation hint
            Icon(
                imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
                contentDescription = null,
                tint = Color.Gray,
                modifier = Modifier.size(20.dp)
            )
        }
    }
}

@Composable
fun SearchResultsList(
    results: List<SearchResult>,
    onSelectResult: (SearchResult) -> Unit,
    modifier: Modifier = Modifier
) {
    LazyColumn(
        modifier = modifier,
        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        items(results, key = { it.doc_id }) { result ->
            SearchResultCard(
                result = result,
                onTap = { onSelectResult(result) }
            )
        }
    }
}

// MARK: - Preview

@Preview(showBackground = true, backgroundColor = 0xFF000000)
@Composable
fun SearchResultCardPreview() {
    val sampleResults = listOf(
        SearchResult(
            doc_id = "1",
            title = "Severe Bleeding Control",
            summary = "Apply direct pressure to the wound immediately. Use a tourniquet if bleeding cannot be controlled with direct pressure. Call emergency services...",
            score = 0.95f,
            module = "critical"
        ),
        SearchResult(
            doc_id = "2",
            title = "Hypothermia Treatment",
            summary = "Move to warm shelter. Remove wet clothing. Insulate the entire body. Give warm beverages if conscious. Seek medical attention...",
            score = 0.87f,
            module = "important"
        ),
        SearchResult(
            doc_id = "3",
            title = "Water Purification Methods",
            summary = "Boil water for at least 1 minute. Use water purification tablets. Filter through clean cloth and sand. UV sterilization in clear bottles...",
            score = 0.72f,
            module = "core"
        )
    )
    
    MaterialTheme {
        SearchResultsList(
            results = sampleResults,
            onSelectResult = { /* Preview */ }
        )
    }
}